packages/@yoda/flora/comp.js

```diff
@@ -1,89 +1,48 @@
-var floraFactory = require('./index')
+'use strict'
 
-module.exports = Flora
-function Flora (logger) {
-  if (logger) {
-    this.__logger = logger
-  } else {
-    this.__logger = {
-      info: () => {},
-      warn: () => {},
-      error: () => {}
-    }
-  }
-  this.__config = {}
-  this.__cli = null
-}
-
-Flora.prototype.handlers = {}
-
-/**
- * Initialize flora client.
- * @param {object} config
- */
-Flora.prototype.init = function init (clientName, config) {
-  this.__config = config
-  this.__logger.info('start initializing flora client')
-  var cli = floraFactory.connect(`${config.uri}#${clientName}`, config.bufsize)
-  if (!cli) {
-    this.__logger.warn('flora connect failed, try again after', config.reconnInterval, 'milliseconds')
-    setTimeout(() => this.init(), config.reconnInterval)
-    return
-  }
-  cli.on('recv_post', this.onRecvPost.bind(this))
-  cli.on('disconnected', this.onDisconnect.bind(this))
-
-  Object.keys(this.handlers).forEach(it => {
-    cli.subscribe(it)
-  })
+var flora = require('@yoda/flora')
 
-  this.__cli = cli
+var defaultConfig = {
+  'uri': 'unix:/var/run/flora.sock',
+  'bufsize': 0,
+  'reconnInterval': 10000
 }
 
-Flora.prototype.destruct = function destruct () {
-  if (this.__cli == null) {
-    return
-  }
-  this.__cli.close()
+function FloraComp () {
 }
 
-/**
- * Flora recv_post channel message handler.
- *
- * @param {string} name
- * @param {string} type
- * @param {string} msg
- */
-Flora.prototype.onRecvPost = function onRecvPost (name, type, msg) {
-  var handler = this.handlers[name]
-  if (handler == null) {
-    this.__logger.error(`No handler found for ${name}`)
-    return
+FloraComp.prototype.init = function (fid, config) {
+  if (typeof fid !== 'string') {
+    fid = ''
+  }
+  if (typeof config !== 'object') {
+    config = defaultConfig
+  }
+  this.agent = new flora.Agent(config.uri + '#' + fid,
+    config.reconnInterval, config.bufsize)
+
+  if (typeof this.handlers === 'object') {
+    Object.keys(this.handlers).forEach((key) => {
+      var cb = this.handlers[key]
+      if (typeof cb === 'function') {
+        this.agent.subscribe(key, cb.bind(this))
+      }
+    })
+    this.agent.start()
   }
-  handler.call(this, msg)
 }
 
-/**
- * Flora disconnection event handler.
- */
```

Comment:
Ditto.

Reply:
no disconnect event.
flora auto reconnect in background.